icu4c/source/common/dictbe.cpp

```diff
@@ -1385,6 +1405,26 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     // inputMap goes out of scope
     return numBreaks;
 }
+
+void CjkBreakEngine::loadJpParticleAndAuxVerbs() {
+    UErrorCode error = U_ZERO_ERROR;
+    const char* tags[2] = {"particles", "auxVerbs"};
+
+    ResourceBundle ja(U_ICUDATA_BRKITR, "ja", error);
+    if (U_SUCCESS(error)) {
+       for(int32_t i = 0 ; i < 2 ; i++) {
+           ResourceBundle bundle = ja.get(tags[i], error);
+           if (U_SUCCESS(error)) {
```

Comment:
ICU api internally follow the check first fail pattern, so you do not need to check U_SUCCESS if there are no dependency other than the success code
I think you can simplify
```
           if (U_SUCCESS(error)) {
               while(bundle.hasNext()){
                   UnicodeString str=bundle.getNextString(error);
                   if(U_SUCCESS(error)){
                       fParticleSet.put(str, (void*)true, error);
                   }
               }
           }
```
to
```
```
               while(U_SUCCESS(error) && bundle.hasNext()){
                   fParticleSet.puti(bundle.getNextString(error), 1, error);
               }
```

Reply:
Thanks. Resolved.